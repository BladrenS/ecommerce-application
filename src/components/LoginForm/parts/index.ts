export { AuthIcon } from './AuthIcon/AuthIcon';
export { AuthInput } from './AuthInput/AuthInput';
