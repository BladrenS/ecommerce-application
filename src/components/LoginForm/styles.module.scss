@use '../../styles/abstract/mixins' as *;

.form {
  display: flex;
  flex-direction: column;
  margin-bottom: 55px;
}

.container {
  display: flex;
  flex-direction: column;
  gap: 15px;
}

.credentials-error {
  color: var(--error);
  text-align: center;
}

.wrapper {
  position: relative;
  top: -82px;

  width: 100%;
  max-width: 395px;
  border-radius: var(--border-radius);

  color: var(--secondary-color);
  letter-spacing: 1.32px;
}

.title {
  margin-bottom: 72px;
  text-align: center;
}

.redirect {
  display: flex;
  gap: 7px;
  justify-content: center;

  color: var(--primary-color);
  text-align: center;
}

.link {
  position: relative;
  color: var(--secondary-color);
  transition: var(--transition);

  @include hover-decor;
}

.green {
  color: green;
}
