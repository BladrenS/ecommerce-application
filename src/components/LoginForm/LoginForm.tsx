import { zodResolver } from '@hookform/resolvers/zod';
import type { FC } from 'react';
import { useForm } from 'react-hook-form';
import { NavLink } from 'react-router-dom';

import { Button, Loader } from '../Ui';
import { useLoginSubmitting } from './logic/useLoginSubmitting';
import { AuthInput } from './parts';
import { type LoginField, schema } from './schemas/loginSchemas';
import styles from './styles.module.scss';

export const LoginForm: FC = () => {
  const {
    register,
    formState: { errors, isValid },
    handleSubmit,
    setError,
  } = useForm<LoginField>({ mode: 'onChange', resolver: zodResolver(schema) });

  const { submit, loading } = useLoginSubmitting(setError);

  return (
    <div className={styles.wrapper}>
      {loading && <Loader />}
      <div className={styles.title}>Login</div>
      <form onSubmit={handleSubmit(submit)} className={styles.form}>
        <div className={styles.container}>
          <AuthInput
            error={errors.email?.message}
            type="email"
            autoComplete="true"
            placeholder="Enter your email"
            {...register('email')}
          />
          <AuthInput
            error={errors.password?.message}
            type="password"
            autoComplete="true"
            placeholder="Enter your password"
            {...register('password')}
          />
        </div>
        <div style={{ textAlign: 'center' }}>
          <Button disabled={!isValid}>Login</Button>
        </div>
        {errors.root && <p className={styles['credentials-error']}>{errors.root.message}</p>}
      </form>
      <div className={styles.redirect}>
        Donâ€™t have an account?
        <NavLink to={'/register'} className={styles.link}>
          Sign Up
        </NavLink>
      </div>
    </div>
  );
};
