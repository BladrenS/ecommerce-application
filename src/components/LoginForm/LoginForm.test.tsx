import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { LoginForm } from './LoginForm';

jest.mock('./logic/useLoginSubmitting', () => ({
  useLoginSubmitting: () => ({
    submit: jest.fn(),
    register: () => ({
      name: 'email',
      onChange: jest.fn(),
      onBlur: jest.fn(),
      ref: jest.fn(),
    }),
    isValid: true,
    loading: false,
    errors: {},
  }),
}));

describe('LoginForm', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <LoginForm />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
