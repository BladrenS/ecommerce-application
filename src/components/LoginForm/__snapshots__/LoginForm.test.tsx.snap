// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="subtitle"
    >
      <div
        class="active"
      >
        Login
      </div>
      <span
        class="decor"
      />
      <a
        class="auth-page"
        data-discover="true"
        href="/register"
      >
        Register
      </a>
    </div>
    <div
      class="title"
    >
      Enter your email and password to login.
    </div>
    <form
      class="form"
    >
      <div
        class="container"
      >
        <label
          class="label"
        >
          <div
            class="name"
          >
            email
          </div>
          <div
            class="wrapper"
          >
            <input
              autocomplete="true"
              class="input"
              name="email"
              placeholder="Enter your email"
              type="text"
            />
          </div>
          <div
            class="error"
          />
        </label>
        <label
          class="label"
        >
          <div
            class="name"
          >
            email
          </div>
          <div
            class="wrapper"
          >
            <input
              autocomplete="true"
              class="input"
              name="email"
              placeholder="Enter your password"
              type="password"
            />
          </div>
          <div
            class="error"
          />
        </label>
      </div>
      <div
        style="text-align: center;"
      >
        <button
          class="button"
        >
          Login
        </button>
      </div>
    </form>
    <div
      class="redirect"
    >
      Don't have an account yet?
      <a
        class="link"
        data-discover="true"
        href="/register"
      >
        Register
      </a>
    </div>
  </div>
</DocumentFragment>
`;
